src/sync.py

```diff
@@ -1,74 +1,91 @@
-                        dest_children.remove(z.jira_key)  # remove it from the list so we can tell if any are left at the end
-
-            for issue in dest_children:  # any issues left in this list do not belong to the epic in ZenHub,
-                sink.remove_from_this_epic(issue)  # so we remove them from the epic in Jira
-
-        elif source.__class__.__name__ == 'JiraIssue' and sink.__class__.__name__ == 'ZenHubIssue':
-            source_children = source.get_epic_children()  # list
-            dest_children = copy.deepcopy(sink.children)  # make a copy to be edited
 
-            if source.issue_type == 'Epic':  # This issue is an epic
-                if sink.issue_type != 'Epic':
-                    sink.promote_issue_to_epic()
+        # Get lists of epic children
+        source_children = source.get_epic_children()
+        sink_children = sink.get_epic_children()
 
-                for issue in source_children:
-                    j = JiraIssue(key=issue, org=source.jira_org)
+        for issue in source_children:  # It could have 0 children
+            source_child = source.repo.issues[str(issue)]  # Get the Issue object by its key
 
-                    if j.github_key not in sink.children:  # issue belongs to this epic in Jira but not ZenHub
-                        z = ZenHubIssue(key=j.github_key, repo=sink.github_repo, org=sink.github_org)
-                        sink.change_epic_membership(add=z.github_key)
+            if source_child.__class__.__name__ == 'ZenHubIssue':  # Get the key of the same issue in the opposite
+                twin_key = source_child.jira_key                  # management system
+            else:
+                twin_key = source_child.github_key
 
-                    else:  # issue belongs to this epic in both Jira and ZenHub
-                        dest_children.remove(j.github_key)  # remove it from the list so we can tell if any are left at the end
+            if twin_key not in sink_children:  # issue belongs to this epic in source but not sink yet,
+                sink.change_epic_membership(add=twin_key)  # so add it as a child of sink
 
-            for issue in dest_children:  # any issues left in this list do not belong to the epic in Jira,
-                z = ZenHubIssue(key=issue, repo=sink.github_repo, org=sink.github_org)  # so we remove them from the epic in ZenHub
-                sink.change_epic_membership(remove=z.github_key)
+            else:  # issue already belongs to this epic in both source and sink; remove it from the list so we can
+                sink_children.remove(twin_key)  # tell if any are left at the end
 
+        for issue in sink_children:              # any issues left in this list do not belong to the epic in source,
+            sink.change_epic_membership(remove=issue)  # so they are removed from the epic in sink.
 
 
+if __name__ == '__main__':
+    a = JiraRepo(repo_name='TEST', jira_org='ucsc-cgl')
```

Comment:
Add a comment that this is specific for the demo execution?

Reply:
oh, this was left over from testing. deleted it.